client/packages/invoices/src/Prescriptions/api/hooks/usePatchState.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * Simple hook to manage a data patch state, which is then passed into a
+ * mutation to update a record.
+ *
+ * It holds the patch state, as well as provides methods to update and reset the
+ * patch, and provides the "dirty" value (i.e. whether or not there is any
+ * difference between the patch and the original "reference" data)
+ */
+
```

Comment:
This is exactly the same in several places, so made a new hook for this. I'll make a follow-up PR (this is already big enough!) where I've moved this into "common" and implemented in other places that follow the same pattern.